cpp/include/cudf/datetime.hpp

```diff
@@ -38,6 +38,23 @@ namespace datetime {
  * @file
  */
 
+/**
+ * @brief Types of datetime components that may be extracted.
+ */
+enum class datetime_component : uint8_t {
```

Comment:
The only thing I can think of that might be causing the problem is this enum because its essentially the only difference between `extract_datetime_component` and the other functions.

Reply:
As soon as we can get the `extract_datetime_component` to properly show up, sphinx should be able to create 
 the hyperlink between the pylibcudf and libcudf functions.